Objective-C/CBLReplicatorConfiguration.m

```
@@ -189,31 +211,95 @@ - (void) setEnableAutoPurge: (BOOL)enableAutoPurge {
     _enableAutoPurge = enableAutoPurge;
 }
 
+- (CBLDatabase*) database {
+    if (!_database)
+        [NSException raise: NSInternalInconsistencyException
+                    format: @"Attempt to access database property, with initWithTarget:"];
+    return _database;
+}
+
 - (void) addCollection: (CBLCollection*)collection
                 config: (nullable CBLCollectionConfiguration*)config {
+    if (!collection.isValid) {
+        [NSException raise: NSInvalidArgumentException
+                    format: @"Attempt to add an invalid collection"];
+    }
     
-    // TODO: Add implementation
-
+    if (_collections.count > 0) {
+        if (_collections[0].db != collection.db) {
+            [NSException raise: NSInvalidArgumentException
+                        format: @"Attempt to add collection from different databases"];
+        }
+    }
+    
+    CBLCollection* defaultCollection = [_database defaultCollectionOrThrow];
+    if (collection == defaultCollection)
+        [self checkAndUpdateConfig: config];
+    
+    CBLCollectionConfiguration* colConfig = nil;
+    if (config)
+        colConfig = [[CBLCollectionConfiguration alloc] initWithConfig: config];
+    else
+        colConfig = [[CBLCollectionConfiguration alloc] init];
+    
+    [_collectionConfigs setObject: colConfig forKey: collection];
+    
+    _collections = _collectionConfigs.allKeys;
 }
 
-- (void) addCollections: (NSArray*)collections
+- (void) addCollections: (NSArray<CBLCollection*>*)collections
                  config: (nullable CBLCollectionConfiguration*)config {
-
-    // TODO: Add implementation
+    Assert(collections.count > 0);
```

Should throw invalid argument exception instead.